packages/cli-platform-apple/src/commands/runCommand/installApp.ts

```diff
@@ -0,0 +1,103 @@
+import child_process from 'child_process';
+import {CLIError, logger} from '@react-native-community/cli-tools';
+import {IOSProjectInfo} from '@react-native-community/cli-types';
+import chalk from 'chalk';
+import {getBuildPath} from './getBuildPath';
+import {getBuildSettings} from './getBuildSettings';
+import path from 'path';
+
+function handleLaunchResult(
+  success: boolean,
+  errorMessage: string,
+  errorDetails = '',
+) {
+  if (success) {
+    logger.success('Successfully launched the app');
+  } else {
+    logger.error(errorMessage, errorDetails);
+  }
+}
+
+type Options = {
+  buildOutput: any;
+  xcodeProject: IOSProjectInfo;
+  mode: string;
+  scheme: string;
+  target?: string;
+  udid: string;
+  binaryPath?: string;
+};
+
+export default async function installApp({
+  buildOutput,
+  xcodeProject,
+  mode,
+  scheme,
+  target,
+  udid,
+  binaryPath,
+}: Options) {
+  let appPath = binaryPath;
+
+  const buildSettings = await getBuildSettings(
+    xcodeProject,
+    mode,
+    buildOutput,
+    scheme,
+    target,
+  );
+
+  if (!appPath) {
+    appPath = await getBuildPath(buildSettings);
+  }
+
+  if (!buildSettings) {
+    throw new CLIError('Failed to get build settings for your project');
+  }
```

Comment:
That's also not the most informative error ever, soon I'll fix all errors in this place :)